clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -123,123 +63,72 @@ export class CopybookDownloadService {
-    paths: string[] | { path: string; profile?: string }[] | undefined,
-  ): Promise<boolean> {
-    if (!paths) return false;
-
-    for (const path of paths) {
-      const p = typeof path === "object" ? path.path : path;
-      const profile =
-        typeof path === "object" && path.profile
-          ? path.profile
-          : ProfileUtils.getProfileNameForCopybook(
-              documentUri,
-              this.explorerApi,
-            );
-      if (profile) {
-        const extensions =
-          await SettingsService.getCopybookExtension(documentUri);
-        const success =
-          downloader instanceof CopybookDownloaderForDsn
-            ? await downloader.downloadCopybook(copybook, p, profile)
-            : await downloader.downloadCopybook(
-                copybook,
-                p,
-                profile,
-                extensions ? extensions : [""],
-              );
-        if (success) return true;
-      }
-    }
-
-    return false;
-  }
-
   private handleAsEndevorElement(documentUri: string) {
     return (
       SettingsService.getCopybookEndevorDependencySettings() ===
         ENDEVOR_PROCESSOR && this.e4eApi?.isEndevorElement(documentUri)
     );
   }
 
-  public makeResolveCopybookHandler() {
-    return this.resolveCopybookHandler.bind(this);
-  }
-
-  public makeCopybookDownloadHandler() {
-    return (
-      cobolFileName: string,
-      copybookNames: CopybookName[],
-      _quietMode: boolean,
-    ) => {
-      return this.downloadCopybooks(cobolFileName, copybookNames);
-    };
+  public makeResolveCopybookUriHandler() {
+    return this.resolveCopybookURI.bind(this);
   }
 
-  public async resolveCopybookHandler(
-    documentUri: string,
+  async resolveCopybookURI(
+    documentURI: string,
     copybookName: string,
     dialectType: string,
   ): Promise<string | undefined> {
-    if (this.handleAsEndevorElement(documentUri)) {
-      const copybookUri = await this.e4eDownloader?.getE4ECopyBookLocation(
+    // is endevor -> download copybook from endevor and return local Uri
+    if (this.handleAsEndevorElement(documentURI)) {
+      const endevorResult = await this.e4eDownloader?.downloadCopybookE4E(
+        documentURI,
         copybookName,
-        documentUri,
       );
-      return copybookUri?.toString();
+
+      return endevorResult?.toString();
     }
+    // search processor groups
     const pgConfigs = await loadProcessorGroupCopybookPathsConfig(
-      { scopeUri: documentUri },
+      { scopeUri: documentURI },
       [],
       dialectType,
     );
-    if (pgConfigs.length > 0) {
-      return (
-        await searchCopybookinProcessorGroups(
-          documentUri,
-          copybookName,
-          this.storagePath,
-          pgConfigs,
-          this.e4eDownloader,
-          this.dsnDownloader,
-          this.ussDownloader,
-        )
-      )?.toString();
+
+    const processorGroupsResult =
+      await this.resolveCopybookUriInProcessorGroups(
```

Comment:
This changes behavior - The new version falls back to settings if PG is used and copybook is not found.